components/icu4x_ecma402/src/pluralrules.rs

```diff
@@ -0,0 +1,163 @@
+//! TODO: License blurb.
+
+//! Implements the traits found in [ecma402_traits::pluralrules].
+
+use ecma402_traits;
+use icu_pluralrules as ipr;
+use icu_pluralrules::PluralRulesError;
+
+pub(crate) mod internal {
+    use ecma402_traits::pluralrules::options::Type;
+    use ecma402_traits::pluralrules::Options;
+    use icu_pluralrules::{PluralCategory, PluralOperands, PluralRuleType};
+
+    /// Converts the trait style option to an equivalent ICU4X type.
+    pub fn to_icu4x_type(style: &Type) -> PluralRuleType {
+        match style {
+            Type::Ordinal => PluralRuleType::Ordinal,
+            Type::Cardinal => PluralRuleType::Cardinal,
+        }
+    }
+
+    /// Converts the number to format into the operands representation.
+    pub fn to_icu4x_operands(n: f64, _opts: Options) -> PluralOperands {
+        // TODO: This does not convert the display Options correctly but
+        // probably should.
+        let nstr = format!("{}", n);
+        nstr.parse().unwrap()
+    }
+
+    /// Expresses the [PluralCategory] as a `str`.
+    pub fn as_str(c: PluralCategory) -> &'static str {
+        match c {
+            PluralCategory::Few => "few",
+            PluralCategory::Many => "many",
+            PluralCategory::One => "one",
+            PluralCategory::Other => "other",
+            PluralCategory::Two => "two",
+            PluralCategory::Zero => "zero",
+        }
+    }
+}
+
+pub struct PluralRules {
+    opts: ecma402_traits::pluralrules::Options,
+    rep: ipr::PluralRules,
+}
+
+impl ecma402_traits::pluralrules::PluralRules for PluralRules {
+    type Error = PluralRulesError;
+
+    fn try_new<L>(l: L, opts: ecma402_traits::pluralrules::Options) -> Result<Self, Self::Error>
+    where
+        L: ecma402_traits::Locale,
+        Self: Sized,
+    {
+        // Which data provider should be the default here?
+        let dp = icu_data_provider::InvariantDataProvider;
+        PluralRules::try_new_with_provider(l, opts, &dp)
+    }
+
+    fn select<W>(&self, number: f64, writer: &mut W) -> std::fmt::Result
+    where
+        W: std::fmt::Write,
+    {
+        let op = internal::to_icu4x_operands(number, self.opts.clone());
+        let result = self.rep.select(op);
+        write!(writer, "{}", internal::as_str(result))
+    }
+}
+
+impl PluralRules {
+    /// Creates a new [PluralRules], using the specified data provider.
+    pub fn try_new_with_provider<L, P>(
+        l: L,
+        opts: ecma402_traits::pluralrules::Options,
+        provider: &P,
+    ) -> Result<Self, PluralRulesError>
+    where
+        L: ecma402_traits::Locale,
+        P: icu_data_provider::DataProvider<'static>,
+        Self: Sized,
+    {
+        let locale: String = format!("{}", l);
```

Comment:
is there no way to make `ecma402_traits::Locale` produce `icu_locale::Locale`?

Reply:
The definitions inside `ecma402_traits` don't have implementations typically, so I suppose that the answer here is no.